activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -26,11 +28,21 @@ module Associations
     class CollectionAssociation < Association #:nodoc:
 
       # Implements the reader method, e.g. foo.items for Foo.has_many :items
-      def reader(force_reload = false)
-        if force_reload
-          klass.uncached { reload }
+      def reader(force_reload = nil, reload: false)
+        unless force_reload.nil?
+          ActiveSupport::Deprecation.warn(<<-MSG.squish)
+            Passing only `true` to force a collection to reload is now
+            deprecated and will be removed in Rails 5.1. Please pass
+            `reload: true` instead.
+          MSG
+
+          reload = force_reload
+        end
+
+        if reload
+          klass.uncached { self.reload }
         elsif stale_target?
-          reload
+          self.reload
```

Comment:
I think we've usually preferred `reload()` over `self.reload` when there's a conflict with a local variable name.